server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -848,6 +848,11 @@ private SegmentIdWithShardSpec createNewSegment(
         versionOfExistingChunk = null;
       }
 
+      // The number of core partitions must always be chosen from the set of used segments in the VersionedIntervalTimeline.
+      // When the core partitions have been dropped, using pending segments may lead to an incorrect state
+      // where the chunk is believed to have core partitions and queries results are incorrect.
+      int numCorePartitions = maxId == null ? 0 : maxId.getShardSpec().getNumCorePartitions();
```

Comment:
The change as written requires an understanding that the `maxId` variable is going to be replaced to understand that this specific position of this line is very important for correctness.  I might recommend that we maintain a `commitedMaxId` and a `overallMaxId` as two separate variables and then use the correct one down at the bottom.  This makes it more clear what the state that we are depending on is.

Reply:
@imply-cheddar , thank you for the review. I've made the changes that you suggested